language/move-stdlib/src/natives/string.rs

```diff
@@ -190,6 +190,57 @@ pub fn make_native_index_of(gas_params: IndexOfGasParameters) -> NativeFunction
     )
 }
 
+/***************************************************************************************************
+ * native fun internal_next_char_boundary
+ *
+ *   gas cost: base_cost + unit_cost * bytes_iterated
+ *
+ **************************************************************************************************/
+#[derive(Debug, Clone)]
+pub struct NextCharBoundaryGasParameters {
+    pub base: InternalGas,
+    pub per_byte: InternalGasPerByte,
+}
+
+fn native_next_char_boundary(
+    gas_params: &NextCharBoundaryGasParameters,
+    _context: &mut NativeContext,
+    _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(args.len() == 2);
+
+    let i = pop_arg!(args, u64) as usize;
+
+    let s_arg = pop_arg!(args, VectorRef);
+    let s_ref = s_arg.as_bytes_ref();
+    let s_str = unsafe {
+        // This is safe because we guarantee the bytes to be utf8.
+        std::str::from_utf8_unchecked(s_ref.as_slice())
+    };
+
+    if i >= s_ref.len() {
+        // TODO: what abort code should we use here?
```

Comment:
Uhuh. That TODO is also in substring, and should have been long resolved. Can you use 2 for now?